src/main/java/mekanism/client/render/item/ChemicalFluidBarDecorator.java

```diff
@@ -0,0 +1,107 @@
+    private final boolean showFluid;
+    private final Predicate<ItemStack> visibleFor;
+
+    /**
+     * @param showFluid if the fluidcapability should be checked for display, display above chemicalCaps if both are present
+     * @param visibleFor checks if bars should be rendered for the given itemstack
+     * @param chemicalCaps the capabilities to be displayed in order, starting from the bottom
+     */
+    @SafeVarargs
+    public ChemicalFluidBarDecorator(boolean showFluid, Predicate<ItemStack> visibleFor, Capability<? extends IChemicalHandler<?, ?>>... chemicalCaps) {
+        this.showFluid = showFluid;
+        this.chemicalCaps = chemicalCaps;
+        this.visibleFor = visibleFor;
+    }
+
+    @Override
+    public boolean render(Font font, ItemStack stack, int xOffset, int yOffset, float blitOffset) {
+        if (!visibleFor.test(stack)) {
+            return false;
+        }
+        RenderSystem.disableDepthTest();
+        yOffset = yOffset + 12;
+        for (Capability<? extends IChemicalHandler<?,?>> chemicalCap : chemicalCaps) {
+            Optional<? extends IChemicalHandler<?, ?>> capabilityInstance = stack.getCapability(chemicalCap).resolve();
+            if (capabilityInstance.isPresent()) {
+                IChemicalHandler<?, ?> chemicalHandler = capabilityInstance.get();
+                if (chemicalHandler.getTanks() == 0) {
+                    continue;
+                }
+                int tank = getDisplayTank(chemicalHandler.getTanks());
+                if (tank != -1) {
+                    renderBarChemical(xOffset, yOffset, chemicalHandler.getChemicalInTank(tank), chemicalHandler.getTankCapacity(tank));
+                    yOffset--;
+                }
+            }
+        }
+
+        if (showFluid) {
+            Optional<IFluidHandlerItem> capabilityInstance = FluidUtil.getFluidHandler(stack).resolve();
+            if (capabilityInstance.isPresent()) {
+                IFluidHandlerItem fluidHandler = capabilityInstance.get();
+                if (fluidHandler.getTanks() != 0) {
+                    int tank = getDisplayTank(fluidHandler.getTanks());
+                    if (tank != -1) {
+                        renderBarFluid(xOffset, yOffset, fluidHandler.getFluidInTank(tank), fluidHandler.getTankCapacity(tank));
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    private void renderBarChemical( int stackXPos, int yPos, ChemicalStack<?> stack, long capacity) {
+        int color = stack.getChemicalColorRepresentation();
+        renderBar(stackXPos, yPos, StorageUtils.getRatio(stack.getAmount(), capacity), color);
+    }
+
+    private void renderBarFluid(int stackXPos, int yPos, FluidStack stack, long capacity) {
+        int color = FluidUtils.getRGBDurabilityForDisplay(stack).orElse(0xFFFFFFFF);
+        renderBar(stackXPos, yPos, StorageUtils.getRatio(stack.getAmount(), capacity), color);
```

Comment:
Given this is the same for renderBarChemical and this method, it might be worth simplifying the method they both call some to have the signature:
```java
private void renderBar(int stackXPos, int yPos, long amount, long capacity, int color)
```
That way they don't need to duplicate the calls to `getRatio`